Allow the use of a custom `overrides.json` file for configuring JupyterLite at runtime

agriyakhetarpal opened the pull request:
## Description

This PR introduces a configuration option in `conf.py` called `jupyterlite_overrides`. This option modifies the command used to build JupyterLite internally within `jupyterlite-sphinx` to use a user-provided `overrides.json` file.

## Rationale

- It is currently possible to leave an `overrides.json` file in the `docs/` directory, similar to [`ipycanvas`'s documentation](https://github.com/jupyter-widgets-contrib/ipycanvas/blob/7226d36cc05e0f15b03af673510ed3e0f3cda424/docs/overrides.json), which will automatically be picked up by `jupyterlite-sphinx` (and thereby JupyterLite) when building the JupyterLite site.
- However, projects using `jupyterlite-sphinx` can benefit from the flexibility of using a custom override file with the docs, which is possible if one has access to the JupyterLite build command with ease.
This option is already available via https://jupyterlite-sphinx.readthedocs.io/en/latest/configuration.html#additional-cli-arguments-for-jupyter-lite-build, which was established via #169. Still, that approach has been directed mostly towards advanced users, and having this feature exposed through a configuration option similar to one for a build-time configuration file would make it easier to use and make it more visible with the rest of the options in the docs.

## Additional context

Please see https://github.com/jupyterlite/jupyterlite/issues/385#issuecomment-2536286348, which is where I got this idea. Some possible use cases for this can be:
- to be able to add download buttons for interactive notebooks within SciPy and PyWavelets that have been/are being established via https://github.com/scipy/scipy/pull/20303/ (cc: @melissawm) and https://github.com/PyWavelets/pywt/pull/741
- customise themes based on the surrounding theme of the page (dark or light – not sure how to switch it yet, but we do have an open issue about it: #124).

agriyakhetarpal commented:
I've (temporarily) pushed bf5e616dbc2cdcb829345c08ab55993cff94db67, which enables the https://github.com/Quansight-Labs/jupyterlab-theme-christmas theme and a download button for notebooks within our documentation. It is working nicely and as expected, and the changes can be tried out in the PR preview: https://jupyterlite-sphinx--225.org.readthedocs.build/en/225/

<img width="1510" alt="A customised JupyterLite deployment within the the Sphinx documentation PR previews for jupyterlite-sphinx where JupyterLite can be seen running with the 'JupyterLab Christmas' theme. A download button is also available for the notebook. " src="https://github.com/user-attachments/assets/9200dcf8-472e-4f22-9533-a7b03bd7eae6" />

Carreau commented on dev-environment.yml:
```diff
@@ -16,3 +16,5 @@ dependencies:
         - .
         - jupyterlite-xeus >=0.2.0a0
+        # Temporary for this PR
+        - jupyterlab_theme_christmas
```
👍

Carreau commented on docs/configuration.md:
```diff
@@ -44,13 +44,27 @@ dependencies:
 
-You can provide [custom configuration](https://jupyterlite.readthedocs.io/en/latest/howto/index.html#configuring-a-jupyterlite-deployment)
-to your JupyterLite deployment.
+You can provide [custom configuration files](https://jupyterlite.readthedocs.io/en/latest/howto/configure/config_files.html)
```
Side question; I know it was already in the readme but do we want to point to latest or stable ?

agriyakhetarpal replied:
Good point, I think pointing to "stable" is better any day. Will change it

agriyakhetarpal replied:
Addressed in 9b241ce8b3f9e9de6361d5505e9e852318ba8505.

Carreau commented on jupyterlite_sphinx/jupyterlite_sphinx.py:
```diff
@@ -772,6 +789,7 @@ def setup(app):
 
     # Config options
     app.add_config_value("jupyterlite_config", None, rebuild="html")
+    app.add_config_value("jupyterlite_overrides", None, rebuild="html")
```
👍

agriyakhetarpal commented:
Rebased on top of `main` and removed the Christmas theme I had added for experimentation. I can confirm that it continues to work with a local build. Thanks again for the review, @Carreau.
